respx/models.py

```diff
@@ -252,26 +254,39 @@ def get_url(self) -> Optional[URLPatternTypes]:
         return self._url
 
     def set_url(
-        self, url: Optional[URLPatternTypes], base: Optional[str] = None
+        self,
+        url: Optional[URLPatternTypes],
+        base: Optional[str] = None,
+        params: Optional[QueryParamTypes] = None,
     ) -> None:
+        params = str(httpx.QueryParams(params))
         url = url or None
         if url is None:
             url = base
+            if base is not None and params:
+                url = url + f"?{params}"
         elif isinstance(url, str):
             url = url if base is None else urljoin(base, url)
             parsed_url = urlparse(url)
             if not parsed_url.path:
                 url = parsed_url._replace(path="/").geturl()
+            if url and params:
+                url = url + f"?{params}"
         elif isinstance(url, tuple):
             url = self.build_url(url)
+            if url and params:
+                url = url + f"?{params}"
         elif isregex(url):
             url = url if base is None else re.compile(urljoin(base, url.pattern))
+            if params:
+                url = re.compile(url.pattern + re.escape(f"?{params}"))
```

Comment:
I like the idea of supporting params adding to given regex.

We should check if regex contains `\?` before joining, and maybe raise an error about not supporting merging regex including querystring with params.